### PR TITLE
Section Styles: Fix ref values within block style variations

### PR DESCRIPTION
**Pull request by aaronrobertshaw**
This PR backports the fix from:
- https://github.com/WordPress/gutenberg/pull/63172

This restores the ability to reference style values from elsewhere within theme.json for block style variations.

Trac ticket: https://core.trac.wordpress.org/ticket/61589#ticket

#### Test Instructions

1. Add some ref values for block style variations to your theme.json e.g. update TT4 with the sections from the theme.json snippet below
2. Load the site editor and edit the default home page
3. Confirm that outline buttons are displayed and have the correct referenced style values
4. Save and ensure the same on the frontend

<details>
<summary>Example theme.json snippet</summary>

```json
{
    "styles": {
        "elements": {
            "button": {
                "border": {
                    "color": "transparent",
                    "radius": "0.25em",
                    "style": "solid",
                    "width": "1px"
                },
                "color": {
                    "background": "var(--wp--preset--color--accent)",
                    "text": "var(--wp--preset--color--base)"
                },
                "spacing": {
                    "padding": {
                        "bottom": "0.5em",
                        "left": "1.25em",
                        "right": "1.25em",
                        "top": "0.5em"
                    }
                }
            }
        },
        "blocks": {
            "core/button": {
                "border": {
                    "color": { "ref": "styles.elements.button.border.color" },
                    "radius": { "ref": "styles.elements.button.border.radius" },
                    "style": { "ref": "styles.elements.button.border.style" },
                    "width": { "ref": "styles.elements.button.border.width" }
                },
                "color": {
                    "background": { "ref": "styles.elements.button.color.background" },
                    "text": { "ref": "styles.elements.button.color.text" }
                },
                "spacing": {
                    "padding": {
                        "bottom": { "ref": "styles.elements.button.spacing.padding.bottom" },
                        "left": { "ref": "styles.elements.button.spacing.padding.left" },
                        "right": { "ref": "styles.elements.button.spacing.padding.right" },
                        "top": { "ref": "styles.elements.button.spacing.padding.top" }
                    }
                },
                "variations": {
                    "outline": {
                        "border": {
                            "color": "currentColor",
                            "width": { "ref": "styles.elements.button.border.width" }
                        },
                        "color": {
                            "text": { "ref": "styles.elements.button.color.background" }
                        },
                        "spacing": {
                            "padding": {
                                "bottom": { "ref": "styles.elements.button.spacing.padding.bottom" },
                                "left": { "ref": "styles.elements.button.spacing.padding.left" },
                                "right": { "ref": "styles.elements.button.spacing.padding.right" },
                                "top": { "ref": "styles.elements.button.spacing.padding.top" },
                            }
                        }
                    }
                }
            }
        }
    }
}
```

</details>

## Screenshots or screencast 

| Before | After |
|---|---|
| Screenshot 2024-07-05 at 6 21 01 PM | Screenshot 2024-07-05 at 6 20 41 PM |

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by andrewserong — APPROVED** [review]
This is testing nicely for me, just as https://github.com/WordPress/gutenberg/pull/63172 does on the site frontend. It'll need the JS packages update from that PR before it'll work in the site editor, but for the PHP changes this change looks good to land from my perspective!

File: src/wp-includes/block-supports/block-style-variations.php [hunk 1]
```diff
@@ -82,6 +118,10 @@ function wp_render_block_style_variation_support_styles( $parsed_block ) {
 		return $parsed_block;
 	}
 
+	// Recursively resolve any ref values with the appropriate value within the
+	// theme_json data.
```
Comment (ramonjd) [inline src/wp-includes/block-supports/block-style-variations.php]:
```suggestion
	/*
	 * Recursively resolve any ref values with the appropriate value within the
	 * theme_json data.
	 */
```

Comment (ramonjd) [reply src/wp-includes/block-supports/block-style-variations.php]: Or single line

```suggestion
	// Recursively resolve any ref values with the appropriate value within the theme_json data.
```

**Review by ramonjd — APPROVED** [review]
I tested along the same lines as https://github.com/WordPress/gutenberg/pull/63172

On the frontend, the values are being resolved from the theme.json tree as expected. 👍🏻 

Screenshot 2024-07-09 at 5 13 37 PM

**Pull request closed without merging**